includes/Handlers/Connection.php

```diff
@@ -1032,4 +1083,260 @@ public function get_plugin() {
+		}
+
+		if ( ! empty( $values->business_manager_id ) ) {
+			$this->update_business_manager_id( sanitize_text_field( $values->business_manager_id ) );
+			$log_data[ self::OPTION_BUSINESS_MANAGER_ID ] = sanitize_text_field( $values->business_manager_id );
+		}
+
+		if ( ! empty( $values->ad_account_id ) ) {
+			$this->update_ad_account_id( sanitize_text_field( $values->ad_account_id ) );
+			$log_data[ self::OPTION_AD_ACCOUNT_ID ] = sanitize_text_field( $values->ad_account_id );
+		}
+
+		if ( ! empty( $values->instagram_profiles ) ) {
+
+			$instagram_business_id = current( $values->instagram_profiles );
+
+			$this->update_instagram_business_id( sanitize_text_field( $instagram_business_id ) );
+			$log_data[ self::OPTION_INSTAGRAM_BUSINESS_ID ] = sanitize_text_field( $instagram_business_id );
+		}
+
+		if ( ! empty( $values->commerce_merchant_settings_id ) ) {
+			$this->update_commerce_merchant_settings_id( sanitize_text_field( $values->commerce_merchant_settings_id ) );
+			$log_data[ self::OPTION_COMMERCE_MERCHANT_SETTINGS_ID ] = sanitize_text_field( $values->commerce_merchant_settings_id );
+		}
+
+		if ( ! empty( $values->pages ) ) {
+
+			$page_id = current( $values->pages );
+
+			try {
+
+				update_option( \WC_Facebookcommerce_Integration::SETTING_FACEBOOK_PAGE_ID, sanitize_text_field( $page_id ) );
+				$log_data[ \WC_Facebookcommerce_Integration::SETTING_FACEBOOK_PAGE_ID ] = sanitize_text_field( $page_id );
+
+				// get and store a current access token for the configured page
+				$page_access_token = $this->retrieve_page_access_token( $page_id );
+
+				$this->update_page_access_token( $page_access_token );
+				$log_data[ self::OPTION_PAGE_ACCESS_TOKEN ] = sanitize_text_field( $page_access_token );
+
+			} catch ( \Exception $e ) {
+
+				if ( $this->get_plugin()->get_integration()->is_debug_mode_enabled() ) {
+					$this->get_plugin()->log( 'Could not request Page Token: ' . $e->getMessage() );
+				}
+			}
+		}//end if
+
+		if ( $this->get_plugin()->get_integration()->is_debug_mode_enabled() ) {
+			$this->get_plugin()->log( 'WebHook User event saved data' );
+			$this->get_plugin()->log( print_r( $log_data, true ) ); //phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_print_r
+		}
+	}
+
+
+	/**
+	 * Register Extras REST API endpoint
+	 */
+	public function init_extras_endpoint() {
+
+		register_rest_route(
+			'facebook/v1',
+			'extras',
+			array(
+				array(
+					'methods'             => array( 'GET', 'POST' ),
+					'callback'            => array( $this, 'extras_callback' ),
+					'permission_callback' => array( $this, 'extras_permission_callback' ),
+				),
+			)
+		);
+	}
+
+
+	/**
+	 * Endpoint permissions
+	 * Woo Connect Bridge is sending the WebHook request using generated key.
+	 *
+	 * @return boolean
+	 */
+	public function extras_permission_callback() {
+
+		add_filter( 'woocommerce_rest_is_request_to_rest_api', '__return_true' );
+
+		$user = apply_filters( 'determine_current_user', null );
```

Comment:
If possible, perhaps using `is_user_logged_in()` instead of the filter directly would be a better fit.

Reply:
I've tried but I got _false_.
This way we are forcing to use the api keys basic auth.